src/components/BookInput.js

```diff
@@ -1,13 +1,16 @@
-import PropTypes from 'prop-types';
 import { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { v4 as uuid } from 'uuid';
+import { addBook } from '../redux/books/booksSlice';
 
-export default function BookInput(props) {
+export default function BookInput() {
+  const dispatch = useDispatch();
   const [title, setTitle] = useState('');
-  const { onSubmit } = props;
+
   const addNewBook = (e) => {
     e.preventDefault();
     if (title !== '') {
-      onSubmit(title);
+      dispatch(addBook({ book: { id: uuid(), title } }));
```

Comment:
- [ OPTIONAL ] I think that the book object should have the properties `id`, `title`, and `author` but you are missing the `author` property :blush:

Reply:
Thank you for your suggestion! 😊 I'll do that